### ISSUE TITLE
Why so many dead containers？MaxContainers is not effect

### ISSUE THREAD

**Issue by hackers365** [kubeedge/kubeedge#2562]
**What happened and what you expected to happen**:
I noticed that there are many dead containers on the node. So I modified the MaxContainers configuration in edgecore.yaml, but it didn't work.
**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
<details><summary>Kubernets Version</summary>

```console
$ kubectl --version
# paste output here
```

</details>

<details><summary>CloudCore Version</summary>

```console
$ cloudcore --version
# paste output here
```

</details>

<details><summary>EdgeCore Version</summary>

```
$ edgecore --version
# paste output here
```

</details>

**Comment by daixiang0** [comment]
Fixed by #2035

**Comment by hackers365** [comment]
> 
> 
> Fixed by #2035

I think edged.go missing invoke StartGarbageCollection func
